src/main/scala/lenthall/exception/ExceptionAggregation.scala

```diff
@@ -0,0 +1,32 @@
+package lenthall.exception
+
+/**
+  * When mixed in an Exception class,
+  * aggregates multiple error messages into the getMessage method.
+  */
+trait MessageAggregation extends Throwable {
+  def message: String
+  def errorMessages: Traversable[String]
+
+  override def getMessage = {
+    s"""$message\n${errorMessages.mkString("\n")}"""
+  }
+}
+
+/**
+  * When mixed in an Exception class,
+  * aggregates multiple throwables into the extended Exception.
+  */
+trait ThrowableAggregation extends MessageAggregation {
+  def message: String
+  def throwables: Traversable[Throwable]
+
+  throwables foreach addSuppressed
```

Comment:
This makes all the `suppressed` exception stacktraces be printed when the exception is thrown / logged / printed.

```
scala> val a = new Exception
a: Exception = java.lang.Exception

scala> val b = new Exception
b: Exception = java.lang.Exception

scala> b.addSuppressed(a)

scala> b.printStackTrace
java.lang.Exception
	at $line4.$read$$iw$$iw$.<init>(<console>:10)
	at $line4.$read$$iw$$iw$.<clinit>(<console>)
	at $line4.$eval$.$print$lzycompute(<console>:7)
        ...
	Suppressed: java.lang.Exception
		at $line3.$read$$iw$$iw$.<init>(<console>:10)
		at $line3.$read$$iw$$iw$.<clinit>(<console>)
		at $line3.$eval$.$print$lzycompute(<console>:7)
		at $line3.$eval$.$print(<console>:6)
		at $line3.$eval.$print(<console>)
		... 28 more
```